packages/site/cms-site/src/sitePreview/pagesRouter/sitePreviewApiHandler.ts

```diff
@@ -0,0 +1,51 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { Scope, SitePreviewParams } from "../SitePreviewUtils";
+
+type GraphQLClient = {
+    setHeader(key: string, value: string): unknown;
+    request<Response, Variables>(document: string, variables?: Variables): Promise<Response>;
+};
+
+async function sitePreviewApiHandler(req: NextApiRequest, res: NextApiResponse, graphQLClient: GraphQLClient) {
```

Comment:
We could also enforce our own GraphQL fetch here. Using a graphql-request like client should allow easier migration. What do you prefer?

Reply:
the whole point of this is easier migration, so I'd use graphql-request-like